binaries/daemon/src/lib.rs

```diff
@@ -1422,10 +1427,40 @@ impl RunningDataflow {
         Ok(())
     }
 
-    async fn stop_all(&mut self, clock: &HLC) {
+    async fn stop_all(
+        &mut self,
+        clock: &HLC,
+        grace_duration: Option<Duration>,
+        reply_tx: Option<Sender<Option<DaemonCoordinatorReply>>>,
+    ) {
         for (_node_id, channel) in self.subscribe_channels.drain() {
             let _ = send_with_timestamp(&channel, daemon_messages::NodeEvent::Stop, clock);
         }
+
+        let running_nodes = self.running_nodes.clone();
+        tokio::spawn(async move {
+            let duration = grace_duration.unwrap_or(Duration::from_millis(500));
+            tokio::time::sleep(duration).await;
+            let mut system = sysinfo::System::new();
+            system.refresh_processes();
+
+            for (node, node_details) in running_nodes.iter() {
+                if let Some(process) = system.process(Pid::from(node_details.pid.clone())) {
+                    process.kill();
+                    warn!(
+                        "{node} was killed due to not stopping within the {:#?} grace period",
+                        duration
+                    )
+                }
+            }
+
+            let reply = DaemonCoordinatorReply::StopResult(Ok(()));
+            if let Some(tx) = reply_tx {
+                let _ = tx
+                    .send(Some(reply))
+                    .map_err(|_| error!("could not send stop reply from daemon to coordinator"));
+            }
```

Comment:
I think we should still send this reply immediately after _sending out_ the `NodeEvent::Stop` messages. The coordinator waits for this reply so delaying it by `grace_duration` could block the coordinator for quite a bit.

The coordinator uses a separate mechanism for waiting until the node is stopped (the `dataflow.reply_senders` field).

Reply:
I see. Got it.

I will update log messages to say that the stop message was send. But the dataflow might still be running until the end of the grace duration